cassandra/deserializers.pyx

```diff
@@ -127,8 +127,8 @@ cdef class DesCounterColumnType(DesLongType):
 
 cdef class DesDateType(Deserializer):
     cdef deserialize(self, Buffer *buf, int protocol_version):
-        cdef double timestamp = unpack_num[int64_t](buf) / 1000.0
```

Comment:
You are right Adam. Both use a double internally and I can reproduce the rounding issue in pure python as well. The reason why we don't have the same behavior with the datetime_from_timestamp function using the python implementation is that we don't cast anything back to integer: the CPython code does it internally with the double passed for the seconds. The CPython code does one additional step than our code, a round() call for the leftover microseconds. I am going to apply this solution. There is one more thing that I need to investigate, which could potentially make the cython and python differ: when the number is exactly halfway between two integers.

https://github.com/python/cpython/blob/master/Modules/_datetimemodule.c#L2154